pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/resourcepack/armor/ArmorTextureEngine.kt

```diff
@@ -0,0 +1,133 @@
+import io.papermc.paper.datacomponent.item.Equippable
+import org.bukkit.entity.Player
+import org.bukkit.persistence.PersistentDataType
+
+object ArmorTextureEngine : PacketListener {
+    val customArmorTexturesKey = pylonKey("custom_armor_textures")
+    val armorSlots = setOf(5, 6, 7, 8)
+
+    @get:JvmStatic
+    @set:JvmStatic
+    var Player.customArmorTextures: Boolean
+        get() = this.persistentDataContainer.getOrDefault(customArmorTexturesKey, PersistentDataType.BOOLEAN, false)
+        set(value) = this.persistentDataContainer.set(customArmorTexturesKey, PersistentDataType.BOOLEAN, value)
+
+    override fun onPacketSend(event: PacketSendEvent?) {
+        if (event == null) return
+
+        val player = event.getPlayer<Player>()
+        if (player == null || !player.customArmorTextures) return
+
+        when (event.packetType) {
+            PacketType.Play.Server.SET_SLOT -> handleSetSlot(WrapperPlayServerSetSlot(event))
+            PacketType.Play.Server.WINDOW_ITEMS -> handleWindowItems(WrapperPlayServerWindowItems(event))
+            PacketType.Play.Server.SET_PLAYER_INVENTORY -> handlePlayerInventorySlot(WrapperPlayServerSetPlayerInventory(event))
+            PacketType.Play.Server.ENTITY_EQUIPMENT -> handleEntityEquipmentPacket(WrapperPlayServerEntityEquipment(event))
+            else -> {}
+        }
+    }
+
+    override fun onPacketReceive(event: PacketReceiveEvent?) {
+        if (event == null) return
+
+        val player = event.getPlayer<Player>()
+        if (player == null) return
+
+        when (event.packetType) {
+            PacketType.Play.Client.CREATIVE_INVENTORY_ACTION -> handleCreativeAction(WrapperPlayClientCreativeInventoryAction(event))
+            else -> {}
+        }
+    }
+
+    private fun handleSetSlot(packet: WrapperPlayServerSetSlot) {
+        if (armorSlots.contains(packet.slot)) {
+            packet.item = addArmorTexture(packet.item)
+        }
+    }
+
+    private fun handleWindowItems(packet: WrapperPlayServerWindowItems) {
+        packet.items.toList().forEachIndexed { index, itemStack ->
+            if (armorSlots.contains(index)) {
+                packet.items[index] = addArmorTexture(itemStack)
+            }
+        }
+    }
+
+    private fun handlePlayerInventorySlot(packet : WrapperPlayServerSetPlayerInventory) {
+        if (armorSlots.contains(packet.slot)) {
+            packet.stack = addArmorTexture(packet.stack)
+        }
+    }
+
+    private fun handleEntityEquipmentPacket(packet: WrapperPlayServerEntityEquipment) {
+        packet.equipment.forEach { equipment ->
+            equipment.item = addArmorTexture(equipment.item)
+        }
+    }
+
+    private fun addArmorTexture(packetStack: ItemStack?) : ItemStack? {
+        if (packetStack == null) return null
+        val stack = SpigotConversionUtil.toBukkitItemStack(packetStack)
+        val item = PylonItem.fromStack(stack)
+        if (item != null && item is PylonArmor) {
```

Comment:
```suggestion
        if (item is PylonArmor) {
```
is check will be false if item is null, so the explicit item != null check is not needed